src/components/Reports/DesignationAccountsReport/DesignationAccountsReport.tsx

```diff
@@ -39,21 +43,60 @@ export const DesignationAccountsReport: React.FC<Props> = ({
     setActiveDesignationAccount,
   ] = useSetActiveDesignationAccountMutation();
 
-  const handleCheckToggle = async (
+  const handleCheckToggle = (
     event: React.ChangeEvent<HTMLInputElement>,
     designationAccountId: string,
   ) => {
-    await setActiveDesignationAccount({
+    const optimisticResponse = true;
+    setActiveDesignationAccount({
       variables: {
         input: {
           accountListId,
           active: event.target.checked,
           designationAccountId,
         },
       },
-    });
+      optimisticResponse: {
+        setActiveDesignationAccount: {
+          id: designationAccountId,
+          __typename: 'SetActiveDesignationAccountResponse',
+          active: event.target.checked,
+        },
+      },
+      update: (cache) => {
+        if (!optimisticResponse) return;
```

Comment:
What is this condition doing? Seems like it's always true so it would always continue.